### PR TITLE
chore: replace `reqwest` with `hyper`

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- replace `reqwest` with `hyper`
- prefer `hyper-rustls` over `hyper-tls`
- fix a bug in `download_snapshot_mainnet`
- removes `h2` dep (large crate implementing both http2 server and client)
- removes `openssl` dep (together with https://github.com/ChainSafe/forest/pull/2140) (native dep)

`reqwest` wraps a lot of unneeded large features like `http2` and does not provide feature to opt-out. 

manually verified with 
`cargo run -p forest-cli -- snapshot fetch`
and
`cargo run -p forest-cli -- --chain calibnet snapshot fetch`

**Reference issue to close (if applicable)**

Closes maybe part of https://github.com/ChainSafe/forest/issues/2088

**Other information and links**

### CODE REVIEW

File: forest/cli/src/cli/snapshot_fetch.rs [hunk 1]
```diff
@@ -102,22 +106,30 @@ async fn snapshot_fetch_mainnet(
...
-    let snapshot_url = snapshot_fetch_config.mainnet.snapshot_url.clone();
-    let snapshot_response = client.get(snapshot_url.clone()).send().await?;
-
-    // Use the redirect if available.
-    let snapshot_url = match snapshot_response
-        .headers()
-        .get("x-amz-website-redirect-location")
-    {
-        Some(url) => url.to_str()?.try_into()?,
-        None => snapshot_url,
```
Comment (hanabi1224) [inline forest/cli/src/cli/snapshot_fetch.rs:116]: The old logic does not create new `snapshot_response` after redirection, it does seem logically correct although it works.

File: Cargo.lock [hunk 2]
```diff
@@ -4255,25 +4234,6 @@ dependencies = [
  "subtle",
 ]
 
-[[package]]
-name = "h2"
```
Comment (hanabi1224) [inline Cargo.lock:4259]: `h2` is gone

File: Cargo.lock [hunk 3]
```diff
@@ -6195,51 +6124,12 @@ dependencies = [
...
-]
-
 [[package]]
 name = "openssl-probe"
 version = "0.1.5"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "ff011a302c396a5197692431fc1948019154afc178baf7d8e37367442a4601cf"
 
-[[package]]
-name = "openssl-sys"
```
Comment (hanabi1224) [inline Cargo.lock:6231]: `openssl` is gone

**Review by connormullett — APPROVED** [review]

File: forest/cli/src/cli/snapshot_fetch.rs (hunk 1 above)
Comment (lemmih) [reply forest/cli/src/cli/snapshot_fetch.rs:116]: The old code didn't use a HEAD request and only used the redirect location for naming the output file. Both approaches (using a HEAD request or not) are fine.

**Review by lemmih — APPROVED** [review]
Looks good.

**Pull request merged**